### PR TITLE
Zarr IO Prototype Module

### PR DESCRIPTION
**Pull request by brandon-neth**
This PR includes prototype implementations of a distributed reader and writer for the Zarr Store protocol. It supports user requests for IO-bound applications using data in this format.

`c-blosc` is used to compress and decompress the chunks, so the included correctness tests start by building c-blosc, then link against it before running the tests.

### CODE REVIEW

**Review by mppf — APPROVED** [review]
Cool! Most of my comments can be addressed with TODO comments if they seem like a big ask.

File: test/library/draft/Zarr/PRECOMP [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+git clone https://github.com/Blosc/c-blosc.git
+cd c-blosc
+mkdir build
+cd build
+cmake ..
+make -j4
+cd ../..
+cp c-blosc/build/blosc/*.so* .
```
[the remark this answers is not in the document]
Comment (brandon-neth) [reply test/library/draft/Zarr/PRECOMP:9]: Do you mean if `c-blosc` is already on the machine running the test or if the `c-blosc` directory has already been cloned? 

Both the `c-blosc` and `.so` files are already in `ZarrTest.cleanfiles`, which I've now changed to be a directory wide `CLEANFILES`. I've added a .gitignore too.

[the remark this answers is not in the document]
Comment (mppf) [reply test/library/draft/Zarr/PRECOMP:9]: Yes I was just concerned with c-blosc already being cloned. I didn't notice `ZarrTest.cleanfiles` in my review.

File: test/library/draft/Zarr/Zarr.chpl [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+
+  coforall loc in Locales do on loc {
+    blosc_init();
+    blosc_set_nthreads(bloscThreads);
+    const hereD = A.localSubdomain();
+    ref hereA = A[hereD];
+
+    const localChunks = getLocalChunks(D, hereD, chunkShape);
+    for chunkIndices in localChunks do {
```
[the remark this answers is not in the document]
Comment (brandon-neth) [reply test/library/draft/Zarr/Zarr.chpl:284]: yes and yes.

**Comment by bradcray** [comment]
I think this could also potentially go into the `modules/packages` directory if you're interested in that.  It'd give it a bit more visibility and make it available in the release (for better or worse).  Even if it's not "done" that characterization could apply to several of the current package modules (and it could be noted in the module docs).

I'm not close enough to this to have an opinion on it, just didn't want you to feel as though it wasn't an option (or to mistakenly think that it would be in the release as-is).

**Pull request merged**